pkg/operator2/oauth.go

```diff
@@ -136,3 +148,173 @@ func (c *osinOperator) handleOAuthConfig(configOverrides []byte) (*corev1.Config
 		},
 	}, nil
 }
+
+func convertProviderConfigToOsinBytes(providerConfig *configv1.IdentityProviderConfig) []byte {
+	var providerConfigBytes bytes.Buffer
+
+	bytesWriter := bufio.NewWriter(&providerConfigBytes)
+	serializer := kubejson.NewYAMLSerializer(kubejson.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	switch providerConfig.Type {
+	case configv1.IdentityProviderTypeBasicAuth:
+		basicAuthConfig := providerConfig.BasicAuth
```

Comment:
Add comment `// TODO we need validation to make sure this cannot be nil (this applies to all IDPs)`